apps/prairielearn/src/pages/instructorInstanceAdminPublishing/components/PublishingForm.tsx

```diff
@@ -0,0 +1,615 @@
+        __csrf_token: csrfToken,
+        __action: 'update_access_control',
+        accessControl: {
+          startDate: data.startDate === '' ? null : data.startDate,
+          endDate: data.endDate === '' ? null : data.endDate,
+        },
+        orig_hash: origHash,
+      };
+
+      const response = await fetch(window.location.pathname, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(requestBody),
+      });
+
+      if (response.ok) {
+        window.location.reload();
+        return;
+      }
+
+      const errorData = await response.json();
+      if (errorData.message) {
+        setErrorMessage(errorData.message);
+      }
+    } catch (error) {
+      console.error('Error updating access control:', error);
+      setErrorMessage('Failed to update access control. Please try again.');
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  const handleAddWeek = async (field: 'startDate' | 'endDate') => {
+    const currentValue = field === 'startDate' ? startDate : endDate;
+    if (currentValue) {
+      const currentDate = Temporal.PlainDateTime.from(currentValue);
+      const newValue = currentDate.add({ weeks: 1 });
+      setValue(field, newValue.toString());
+      // setValue doesn't seem to trigger validation so we need to trigger it manually
+      await trigger('startDate');
+      await trigger('endDate');
+    }
+  };
+
+  // Validation
+  const validateStartDate = (value: string) => {
+    if (selectedStatus === 'publish_scheduled') {
+      if (!value) {
+        return 'Start date is required for scheduled publishing';
+      }
+      // Check if start date is in the future
+      const startDateTime = plainDateTimeStringToDate(value, courseInstance.display_timezone);
+      if (startDateTime <= nowDateInTimezone(courseInstance.display_timezone)) {
+        return 'Start date must be in the future for scheduled publishing';
+      }
+    }
+    return true;
+  };
+
+  const validateEndDate = (value: string) => {
+    if (selectedStatus !== 'unpublished') {
+      if (!value) {
+        return 'End date is required';
+      }
+      // Check if end date is after start date
+      if (startDate && value) {
+        const startDateTime = plainDateTimeStringToDate(startDate, courseInstance.display_timezone);
+        const endDateTime = plainDateTimeStringToDate(value, courseInstance.display_timezone);
+        if (endDateTime <= startDateTime) {
+          return 'End date must be after start date';
+        }
+      }
+    }
+    return true;
+  };
+
+  if (hasAccessRules) {
+    return (
+      <div class="alert alert-warning" role="alert">
+        <strong>Legacy Access Rules Active:</strong> This course instance is using the legacy
+        allowAccess system. To use the new access control system, you must first remove all
+        allowAccess rules from the course configuration.
+      </div>
+    );
+  }
```

Comment:
It seems weird to me to render this whole component at all in this case. Would it be better to move this into the legacy component and conditionally render either the legacy one or `PublishingForm` depending on which one should be shown?